### ISSUE TITLE
细节小疑问

### ISSUE THREAD

**Issue by yBeOne** [KikimorMay/MultiFace#7]
你好，我有几个小问题想问一下：
(1)论文给出训练阶段4个head的梯度，后期基本上一致，是否说明，4个head的作用完全一致呢，即提取的特征完全一样呢？
(2)为什么训练时给出的裕量参数m比arcface小0.2？
(3)能给出训练的显卡配置吗？因为特别占用显卡资源
(4)我用较大数据集测试时，2epoch还未收敛，损失值不变，能给出你的训练记录吗？
希望能解答一下，谢谢！！！

**Comment by KikimorMay** [comment]
你好，
(1)是的，在训练结束之后，我们分析过4个head的对应特征，发现它们是高度相关的。同时在multiface训练之后，利用一个单独head对应的sub-feature也能得到将4个concate之后的精度。
(2)我们认为，将总的feature分为小特征之后这个操作已经能够使类内更加紧密了（原文Figure 7，使用softmax loss with without m），所以当使用arcface的时候，m可以对应小一些。
(3)在mobilefacenet中，我们使用的单卡p40，在resnet-ir-se中使用的是单卡V100，如果需要多卡训练，可以参考。[多卡训练](https://github.com/KikimorMay/MultiFace/tree/multicards)
(4)我们提供了训练记录在，你可以进行参考. [mobilefacenet](https://drive.google.com/file/d/16zxRC2t7Bi0GzOeiWfXjBW39V1nxd5TV/view?usp=sharing)

**Comment by yBeOne** [comment]
4个head对应128维，是不是就说明其实用128维来表示人脸特征完全够用了，如果对4个head加上不确定性学习不一样的特征，是不是泛化效果会更好。

**Comment by KikimorMay** [comment]
1）但是单独使用128维去训练，在一些情况下会使得网络优化困难，精度也不够高。我们猜测是4个head共同传递回去的梯度共同优化了网络，使网络训练稳定，精度更高。（我的理解这还是一个优化问题，比如128维足够了，但是我们很难把它优化好）
2）我们尝试着对4个head加上不相关的约束，但是没有得到很好的效果（比如利用对抗去进行训练），但是这个方向直觉上应该是可行的。

**Comment by yBeOne** [comment]
官方推荐用torch.nn.parallel.DistributedDataParallel进行多卡训练
